### PR TITLE
:sparkles: chore: remove gorilla mux | api verbs

### PR DESCRIPTION
**Pull request by perebaj**
A really important principle about REST APIs: they were created to follow some **rules** about how to manage and send data through the web. For this reason, it's important to understand simple concepts, **like Verbs, Status Codes, and how to use them.**

Here you can find some useful notes about it: [API Rest My Way](https://github.com/perebaj/jjhandbook/blob/main/notes/restapi.md)

Even being "my way", I just get it from the internet and put it here to help me to remember it. I hope it can help you too.

### CODE REVIEW

File: routes/routes.go [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", handlers.Index)
 
-	log.Println("Server listening on:8000")
-	http.ListenAndServe(":8000", r)
+	mux.HandleFunc("/newProduct", handlers.NewProductHandler)
+	mux.HandleFunc("POST /products", handlers.InsertProductHandler)
+	mux.HandleFunc("DELETE /products", handlers.DeleteProductHandler)
+	mux.HandleFunc("PATCH /products", handlers.EditProductHandler)
+	mux.HandleFunc("PUT /products", handlers.UpdateProductHandler)
```
Comment (perebaj) [inline routes/routes.go:17]: For example here. Your main entity are **Products**, and even playing around with different actions upon them, the route that will be called is always the same: **/products**, the only thing that changes is the HTTP method, as explained in the table below:

| HTTP Method | Route | Description |
| --- | --- | --- |
| GET | /products | List all products |
| POST | /products | Create a new product |
| GET | /products/:id | Get a product |
| PUT | /products/:id | Update a product |
| DELETE | /products/:id | Delete a product |
| PATCH | /products/:id | Partially update a product |

Comment (perebaj) [reply routes/routes.go:17]: And I also saw that you are using some Path values to change things about the product, like: `/products/{id}` and other values, **like name, description, and so on**

I think that for the POST method, there is a better way to do that. But I will give you the pleasure of discovery it kkkk

Comment (perebaj) [reply routes/routes.go:17]: And I also saw that you are using some Path values to change things about the product, like: `/products/{id}` and other values, **like name, description, and so on**

I think that for the POST method, there is a better way to do that. But I will give you the pleasure of discovery it kkkk

Comment (JoaoGumiero) [reply routes/routes.go:17]: Oh, I'll take a look at your document right away. I owe you one, tks.

**Pull request merged**